### ISSUE TITLE
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\ node_modules\\npm\\bin\\npm-cli.js" "-g" "install" "sails"

### ISSUE THREAD

**Issue by chauragade** [npm/npm#18714]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
I am trying to install sails and it showing following error:

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\
node_modules\\npm\\bin\\npm-cli.js" "-g" "install" "sails"
npm ERR! node v6.11.3
npm ERR! npm  v3.10.10

npm ERR! Invalid name: "@sailshq/express"
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\ychauragade\npm-debug.log

#### How can the CLI team reproduce the problem?
npm -g install sails

### supporting information:

 - `npm -v` prints:
 - `node -v` prints:
 - `npm config get registry` prints:
 - Windows, OS X/macOS, or Linux?:
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by kenany** [comment]
Does `C:\Users\ychauragade\package.json` exist?

**Comment by meldsza** [comment]
Can you include your debug log `C:\Users\ychauragade\npm-debug.log`

**Comment by Butterfly** [comment]
@KenanY 's comment made the path stand out. That seems... I don't know... not the norm, *but* normal doesn't 100% exist when we can all personalize what we're doing.

@meldsza 's comment helps highlight that same path so now I'm wondering. Is that normally where that `npm-debug.log` would *also* land, i.e. directly under the user's home directory instead of inside a deeper `npm` specific directory?

I'm sorry to have to ask instead of knowing. I don't know how to cause a fail to have that appear and then be able to know from firsthand experience.

On a whim, I went looking for just `debug.log` files and found those stored under my user's hidden `.npm` directory. That seems... totally *logical* and is part of why I decided to step out on that thin limb and ask out loud. :smile:
